### ISSUE TITLE
CA certificate management or how to install root certificates

### ISSUE THREAD

**Issue by baimafeima** [getsolus/help-center-docs#95]
I think it'll be important to offer a guide how to install CA/root certificates. Whether you work in a company or you need to connect to a university network, you will need to deal with installing or configuring root certificates. 
- What is a CA/root certificate and why is it important? 
- Where are CA certificates locally stored in Solus?
- How to import CA certificates into openconnect to safeguard the VPN connection?
- Many CA certificates cannot be installed via a web browser or a WiFi connection and in Solus you will get the message "This certificate is already installed as a certificate authority" upon clicking a .crt file in the browser when in fact it is **not** installed yet.

**Comment by JoshStrobl** [comment]
This can probably go under Configuration.

**Comment by DiogoFerrao** [comment]
Could really use a little help, particularly with the last topic, im a linux noob in general, just installed solus and I'm really enjoying, but my college runs a root CA to access the wi fi and without it working i'll need to distro hop

**Comment by parkingmeter** [comment]
I was able to figure this out, generally following the process from https://getsol.us/forums/viewtopic.php?t=1973

I added a self-signed root CA from my pfSense installation on my local machine.

Overview:

1. Validate the file is in .pem format: `openssl x509 -text -noout -in my-root-ca.crt`
2. If that fails, it might be in .der format, you can check with `openssl x509 -in my-root-ca.crt -inform der -text`
3. You can convert der to pem format with `openssl x509 -text -in my-root.crt -inform der -outform pem > ./my-root.pem`
4. Keep the decoded output of the cert for later.
5. Copy the cert `cp my-root.pem /etc/ssl/certs` (I renamed my .crt to .pem)
6. Run `c_rehash` this will create a .0 soft link to the .pem file
7. The linked instructions have you overwrite the ca-certificates.crt file with `cat /etc/ssl/certs/*.0 > /etc/ssl/certs/ca-certificates.crt` but even when running this with sudo I got permission denied errors on ca-certificates.crt. Instead, I manually editied ca-certificates.crt and appended the decoded cert information from above as well as the encoded cert `cat my-root.pem` to match the format for the existing certs in the file.
8. I restarted Chrome and the SSL connection was validated!

I hope this helps!

**Comment by poVoq** [comment]
> but even when running this with sudo I got permission denied errors on ca-certificates.crt

Same issue here... any specific reason for this unexpected behaviour?

**Comment by parkingmeter** [comment]
Finally figured this out...."[Output redirection (via the > operator) is done by the shell, not by echo. You have to login as root](https://askubuntu.com/a/230482)"

The critical step here is that `cat /etc/ssl/certs/*.0 > /etc/ssl/certs/ca-certificate.crt` must be run as root, not via sudo.

**Comment by SirZazu** [comment]
I encounter the same issue recently. The @parkingmeter solution did not work for me as well, but I found out that adding the cert to the user database allows chrome to work with the self-signed cert. The command as non-root is: 

` certutil -d sql:$HOME/.pki/nssdb -A -t "P,," -n <cert_label> -i <path_to_crt_file>`

**Comment by parkingmeter** [comment]
@DataDrake  I revisited this issue after seeing another question about this in the form post from three years ago.

As this has lived in the backlog since then, I'm curious if I can help with writing the documentation? I've cloned this repo and I'll follow the instructions, but I've never submitted a PR for Solus help center docs. Just didn't want to waste time if this help isn't actually wanted ;)
